src/main/app/common/country.ts

```diff
@@ -0,0 +1,122 @@
+import { Form, FormValidationError } from 'forms/form'
+import { ValidationError } from 'class-validator'
+import { PartyDetails } from 'forms/models/partyDetails'
+import * as config from 'config'
+import { request } from 'client/request'
+import { PostcodeInfoClient, PostcodeInfoResponse } from '@hmcts/postcodeinfo-client'
+
+const postcodeClient = new PostcodeInfoClient(config.get<string>('postcodeLookup.apiKey'), request)
+
+export class PostcodeInfoCountry {
+  readonly GSS_CODE: string
+  readonly name: string
+}
+
+export class Country {
+  static readonly WALES = new Country('wales', 'Wales')
+  static readonly ENGLAND = new Country('england', 'England')
+  static readonly SCOTLAND = new Country('scotland', 'Scotland')
+  static readonly NORTHERN_IRELAND = new Country('northernIreland', 'Northern Ireland')
+
+  readonly value: string
+  readonly name: string
+
+  constructor (value: string, name: string) {
+    this.value = value
+    this.name = name
+  }
+
+  static all (): Country[] {
+    return [
+      Country.WALES,
+      Country.ENGLAND,
+      Country.SCOTLAND,
+      Country.NORTHERN_IRELAND
+    ]
+  }
+
+  static defendantCountries (): Country[] {
+    return [
+      Country.WALES,
+      Country.ENGLAND
+    ]
+  }
+
+  static async isValidClaimantAddress (form: Form<PartyDetails>): Promise<Form<PartyDetails>> {
```

Comment:
The method name is misleading. Methods starting with `is` are supposed to answer question in yes / no or true / false way. That method in turn populates validation errors on provided form.